code/modules/hydroponics/grown.dm

```diff
@@ -38,6 +39,13 @@
 
 	transform *= TransformUsingVariable(seed.potency, 100, 0.5) //Makes the resulting produce's sprite larger or smaller based on potency!
 
+	var/datum/plant_gene/trait/glow/G = seed.get_gene(/datum/plant_gene/trait/glow)
+	if(G)
+		if(istype(src.loc,/mob))
+			pickup(src.loc)//adjusts the lighting on the mob
+		else
+			src.SetLuminosity(G.get_lum(seed))
+

```

Comment:
This whole thing needs to be moved to mutation action
Add on_new proc to mutations and try to call it on all mutations in a plant when its created
Place the lighting setup into that proc of the glow mutation